robot-code/micro-python/sumobot.py

```diff
@@ -10,24 +16,32 @@ def __init__(self, robot_name="default", ssid="WATCHTOWER", password="lancerrobo
         self.robot_name = robot_name
         self.my_game_state = 0
         self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        self.socket.settimeout(1.0)
 
         wlan = network.WLAN(network.STA_IF)
         wlan.active(True)
         wlan.connect(ssid, password)
         while not wlan.isconnected():
             time.sleep(1)
+            
         print('network config:', wlan.ifconfig())
-        print("Connected to Wi-Fi. IP:", wlan.ifconfig()[0])
+        print("Connected to Wi-Fi. IP: ", wlan.ifconfig()[0])
+
         self.socket.bind(('', port))
-        connect_to_station()
         print('Receiving on UDP Port...')
 
     ### Called by user to receive the data from UDP
     ### The output either will show the updated game state or the controller status depending on game state
     def read_udp_packet(self):
         data = None
-        while data is None:
+        name = "!" + self.robot_name + "@"
+        self.socket.sendto(name.encode(), ("255.255.255.255", 2367))
```

Comment:
Separate broadcast or robot name from read_udp_packet(), the read will need to be run much more often than the tx in order to consume packets in the buffer that aren't for this robot.